### PR TITLE
[WIP] feat: Upload User Profile Picture

### LINKED ISSUE #1182: Upload User Profile Picture from Local Storage
### Feature Request

**Feature Description**

The user should be able to upload images from device local storage to the server and use the returned URL to show the profile picture in the app.
   
**Parent Issue**

#1093 

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #1182 

**Not ready for review yet**

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream development branch.

Changes: 
1. Request permission to access storage if not given already.
2. User gets option to select images from File Explorer or local Gallery app.
3. The Image is converted to Base64 and sent to server.
4. The url returned by server is used to update `user.avatarUrl` in database and further used by glide to load image.

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/core/organizer/detail/OrganizerDetailPresenter.java [hunk 1]
```diff
@@ -1,26 +1,40 @@
 package org.fossasia.openevent.app.core.organizer.detail;
 
+import android.content.ContentResolver;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/organizer/detail/OrganizerDetailPresenter.java]: No android stuff in presenters

**Comment by Masquerade0097** [comment]
@iamareebjamal There is a bug in server. The url which is returned by server sows error on opening in browser and therefore Glide shows `Failed to load resource` when it's loaded through it.

Example url,  
https://open-event-api-dev.herokuapp.com/static/media/temp/images/9afcf54b-0592-47d9-a1d8-1b6349b04176/emdHRkNCOU/bfcc5ef6-5489-414f-aba1-73e26a9c7100.jpg

I'll create an issue in server for the incorrect urls. Other than that, please review.

File: app/src/main/java/org/fossasia/openevent/app/core/organizer/detail/OrganizerDetailFragment.java [hunk 2]
```diff
@@ -58,6 +69,41 @@ public void onStart() {
...
+            String type = UploadUtils.getMimeTypeFromUri(contentResolver, targetUri);
+
+            try {
+                Bitmap bitmap = BitmapFactory.decodeStream(contentResolver.openInputStream(targetUri));
+                Bitmap resizedBitmap = Bitmap.createScaledBitmap(bitmap, 500, 500, false);
+                String imageString = UploadUtils.convertBitmapToString(resizedBitmap, type);
+
+                getPresenter().uploadImage(imageString);
+            } catch (FileNotFoundException e) {
+                Timber.d(e);
```
Comment (srv-twry) [inline app/src/main/java/org/fossasia/openevent/app/core/organizer/detail/OrganizerDetailFragment.java]: May be we could also show a snackbar saying that "There was a problem with the image" or equivalent ?

**Comment by iamareebjamal** [comment]
@Masquerade0097 Status?

**Comment by iamareebjamal** [comment]
@Masquerade0097 Status?

**Comment by Masquerade0097** [comment]
@iamareebjamal Apologies for the delay, i was facing some errors earlier mentioned above so kept it on hold. I will try my best to finalize it in 1 or 2 days.

**Comment by iamareebjamal** [comment]
The goal of GSoC is to get permanent addition to teams in Open Source projects. It's sad to see everyone leave after the party when there is time when we need people. Although GSoC is over and it is completely up to you guys if you want to continue contributing to the project, but I believe that it would be beneficial for both you and the project if you continue to add support to it at your own pace.

This PR is open for days. Anyone up to finalize it @Masquerade0097 @sridharjajoo @mishuvs

**Comment by Masquerade0097** [comment]
@iamareebjamal  Apologies, I'm very much occupied and won't be able to fix it for now. If anyone else want to work on it then they can.

**Pull request closed without merging**